iceoryx_hoofs/test/moduletests/test_unix_domain_sockets.cpp

```diff
@@ -69,45 +75,282 @@ class UnixDomainSocket_test : public Test
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkTooLongSocketNameWithPathPrefixLeadsToInvalidChannelNameError)
+{
+    UnixDomainSocket::UdsName_t longSocketName;
+    for (uint64_t i = 0U; i < UnixDomainSocket::LONGEST_VALID_NAME - strlen(UnixDomainSocket::PATH_PREFIX) + 1; ++i)
+    {
+        longSocketName.append(cxx::TruncateToCapacity, "o");
+    }
+    auto ret = UnixDomainSocket::unlinkIfExists(longSocketName);
+    ASSERT_TRUE(ret.has_error());
+    EXPECT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+}
+
+TEST_F(UnixDomainSocket_test, UnlinkExistingSocketIsSuccessful)
+{
+    UnixDomainSocket::UdsName_t socketFileName = UnixDomainSocket::PATH_PREFIX;
+    socketFileName.append(cxx::TruncateToCapacity, "iceoryx-hoofs-moduletest.socket");
+    createTestSocket(socketFileName);
+    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, socketFileName);
+    EXPECT_FALSE(ret.has_error());
 }
 
-TEST_F(UnixDomainSocket_test, UnlinkNonExistingWithInvalidNameLeadsToError)
+TEST_F(UnixDomainSocket_test, UnlinkExistingSocketWithPathPrefixLeadsIsSuccessful)
 {
-    auto ret = UnixDomainSocket::unlinkIfExists(UnixDomainSocket::NoPathPrefix, invalidName);
-    EXPECT_TRUE(ret.has_error());
-    ASSERT_THAT(ret.get_error(), Eq(IpcChannelError::INVALID_CHANNEL_NAME));
+    UnixDomainSocket::UdsName_t socketFileName = "iceoryx-hoofs-moduletest.socket";
+    UnixDomainSocket::UdsName_t socketFileNameWithPrefix = UnixDomainSocket::PATH_PREFIX;
+    socketFileNameWithPrefix.append(cxx::TruncateToCapacity, socketFileName);
+    createTestSocket(socketFileNameWithPrefix);
+    auto ret = UnixDomainSocket::unlinkIfExists(socketFileName);
+    EXPECT_FALSE(ret.has_error());
 }
 
-TEST_F(UnixDomainSocket_test, SendingOnServerLeadsToError)
+void sendOnServerLeadsToError(const sendCall_t& send)
 {
     cxx::string<10> message{"Foo"};
-    Duration duration{1_ms};
-    auto result = server.timedSend(message, duration);
+    auto result = send(message);
     EXPECT_TRUE(result.has_error());
     ASSERT_THAT(result.get_error(), Eq(IpcChannelError::INTERNAL_LOGIC_ERROR));
 }
 
-TEST_F(UnixDomainSocket_test, ReceivingOnClientLeadsToError)
+TEST_F(UnixDomainSocket_test, TimedSendOnServerLeadsToError)
 {
-    Duration duration{1_ms};
-    auto result = client.timedReceive(duration);
+    sendOnServerLeadsToError([&](auto& msg) { return server.timedSend(msg, 1_ms); });
+}
+
+TEST_F(UnixDomainSocket_test, SendOnServerLeadsToError)
+{
+    sendOnServerLeadsToError([&](auto& msg) { return server.send(msg); });
+}
+
+void successfulSendAndReceive(const std::string& message, const sendCall_t& send, const receiveCall_t& receive)
+{
+    ASSERT_FALSE(send(message).has_error());
+
+    auto receivedMessage = receive();
+    ASSERT_FALSE(receivedMessage.has_error());
+    EXPECT_EQ(message, *receivedMessage);
```

Comment:
Theoretically we do not yet know if our abstraction is using the sockets. It's possible to use a global variable for data transfer. Can we have one test using the POSIX API for send and receive directly?

Reply:
You are right, this would make send ... (for all posix wrapper :( I see the dawn of a new issue)

Reply:
It has to be done once, though. When we know that e.g. send works, we can use that to test timedReceive, etc.